packages/core/src/style-decoder.ts

```diff
@@ -175,7 +176,9 @@ export class StyleDecoder {
     const sorted = sortStyleRules(details)
     sorted.forEach((value) => {
       if (value.conditions) {
-        const path = basePath.concat(value.conditions.map((c) => c.rawValue ?? c.raw))
+        const path = basePath.concat(
+          value.conditions.map((c) => this.context.utility.tokens.resolveReference(c.rawValue ?? c.raw)),
+        )
```

Comment:
We could create a `resolveCondition` method in the style decoder

this.context.utility.tokens.resolveReference(c.rawValue ?? c.raw)),`